src/include/access/tableam.h

```diff
@@ -53,6 +53,18 @@ struct ValidateIndexState;
 #define SCAN_SUPPORT_COLUMN_ORIENTED_SCAN (1 << 0)  /* support column-oriented scanning*/
 #define SCAN_SUPPORT_RUNTIME_FILTER       (1 << 1)  /* support runtime filter scan */
 
+/*
+ * Flags represented the supported features of scan.
+ * 
+ * Reserve first 8 bits for public.
+ * 
+ * NOTE: If you add a new flag, make sure the flag's bit is consecutive with
+ * the previous one.
+ */
+#define SCAN_SUPPORT_VECTORIZATION (1 << 8) /* support vectorization scan */
+
+#define SCAN_FORCE_BIG_WRITE_LOCK (1 << 9) /* force big write lock */
+
```

Comment:
The flags should be compact. The lower bits are used by kernel, higher bits are used by other modules or future.